content/en/serverless/google_cloud_run/_index.md

```diff
@@ -2,201 +2,296 @@
+ENV DD_SERVICE=datadog-demo-run-ruby
+ENV DD_ENV=datadog-demo
+ENV DD_VERSION=1
+
+# change the entrypoint to wrap your application into the Datadog serverless-init process
 ENTRYPOINT ["/app/datadog-init"]
+
+# execute your binary application wrapped in the entrypoint. Adapt this line to your needs
 CMD ["rails", "server", "-b", "0.0.0.0"] (adapt this line to your needs)
 
 ```
 
-See [Tracing Ruby Applications][1] for detailed instructions. [Sample code for a simple Ruby application][2].
+[Sample code for a simple Ruby application][1].
+More details on the [Ruby tracing library][2].
+
+[1]: https://github.com/DataDog/crpb/tree/main/ruby-on-rails
+[2]: /tracing/trace_collection/dd_libraries/ruby/?tab=containers#instrument-your-application
 
-[1]: /tracing/trace_collection/dd_libraries/ruby/
-[2]: https://github.com/DataDog/crpb/tree/main/ruby-on-rails
 {{< /programming-lang >}}
 {{< /programming-lang-wrapper >}}
 
-#### Troubleshooting
-This integration depends on your runtime having a full SSL implementation. If you are using a slim image for Node, you may need to add the following command to your Dockerfile to include certificates.
+#### 1.b Instrument using buildpack
 
-```
-RUN apt-get update && apt-get install -y ca-certificates
-```
+[`Pack Buildpacks`][4] provide a convenient way to package your container without using a Dockerfile. This example uses the GCP container registry and Datadog serverless buildpack.
 
-#### Build with the Datadog buildpack
+**Note**: Follow the instructions to install the [tracing library][2] for your language before running the buildpack.
 
-1. Use [`pack`][4] to build your application by running the following:
-   ```
+Build your application by running the following command:
+
+   ```shell
    pack build --builder=gcr.io/buildpacks/builder \
    --buildpack from=builder \
-   --buildpack datadog/serverless-buildpack:beta4 \
+   --buildpack datadog/serverless-buildpack \
    gcr.io/YOUR_PROJECT/YOUR_APP_NAME
-
    ```
 
-   **Note**: Not compatible with Alpine.
+**Note**: Not compatible with Alpine.
 
- 2. Push your image to GCP:
-    ```
-    docker push gcr.io/YOUR_PROJECT/YOUR_APP_NAME
-    ```
+### 2. Datadog Agent configuration
 
-### Deploy to Cloud Run
-Below are instructions for deploying a Cloud Run service using standard GCP tools. If you have other systems in place for managing container images, secrets, or deployments, you may use those instead.
-
-3. Run this command to submit your build to GCP.
-
-   ```
-   gcloud builds submit --tag gcr.io/YOUR_PROJECT/YOUR_APP_NAME
-   ```
-4. Create a secret from your Datadog API key.
-   Go to [Secret Manager][5] in your GCP console and click on **Create secret**.
+Once the container is built and pushed to your registry, the last step is to set the required environment variables for the  Datadog Agent:
+- `DD_API_KEY`: Datadog API key, used to send data to your Datadog account. It should be configured as a GCP Secret for privacy and safety issue.
```

Comment:
Maybe I missed it, but could we keep the link to google's doc about creating and using secret in cloud run?

Reply:
oh my bad, I must have removed all the section. Adding it back